arelle/ModelValue.py

```diff
@@ -157,37 +188,35 @@ def clarkNotation(self) -> Optional[str]:
     def expandedName(self) -> str:
         return '{0}#{1}'.format(self.namespaceURI or "", self.localName)
 
-    def __repr__(self):
+    def __repr__(self) -> str:
         return self.__str__()
 
-    def __str__(self):
+    def __str__(self) -> str:
         if self.prefix:
             return '{}:{}'.format(self.prefix, self.localName)
         else:
+            assert self.localName is not None
             return self.localName
 
-    def __eq__(self, other):
+    def __eq__(self, other: QName) -> bool: # type: ignore[override]
```

Comment:
This typing is wrong I believe. For the lt too? It only make sense to compare with a qname, but there is nothing that actually prevents you from comparing against other objects which is why the conforms method exists

Reply:
Yes, is wrong (and in other places too). I'll fix as per @austinmatherne-wk's suggestions.